lib/service/modules/entity.js

```diff
@@ -0,0 +1,185 @@
+      getWriters: () => _.get(_entity, ['_acl', 'w'], []),
+      getReaderGroups: () => _.get(_entity, ['_acl', 'groups', 'r'], []),
+      getWriterGroups: () => _.get(_entity, ['_acl', 'groups', 'w'], []),
+      addReader: (userId) => {
+        if (!(_.get(_entity, ['_acl', 'r']))) _entity._acl.r = [];
+        if (_entity._acl.r.indexOf(userId) === -1) _entity._acl.r.push(userId);
+        return aclFunctions;
+      },
+      addWriter: (userId) => {
+        if (!(_.get(_entity, ['_acl', 'w']))) _entity._acl.w = [];
+        if (_entity._acl.w.indexOf(userId) === -1) _entity._acl.w.push(userId);
+        return this;
+      },
+      addReaderGroup: (groupId) => {
+        if (!(_.get(_entity, ['_acl', 'groups']))) _entity._acl.groups = [];
+        if (!(_.get(_entity, ['_acl', 'groups', 'r']))) _entity._acl.groups.r = [];
+        if (_entity._acl.groups.r.indexOf(groupId) === -1) _entity._acl.groups.r.push(groupId);
+        return aclFunctions;
+      },
+
+      addWriterGroup: (groupId) => {
+        if (!(_.get(_entity, ['_acl', 'groups']))) _entity._acl.groups = [];
+        if (!(_.get(_entity, ['_acl', 'groups', 'w']))) _entity._acl.groups.w = [];
+        if (_entity._acl.groups.w.indexOf(groupId) === -1) _entity._acl.groups.w.push(groupId);
+        return aclFunctions;
+      },
+      removeReader: (userId) => {
+        let readers = _.get(_entity, ['_acl', 'r'], []);
+        let userIx = readers.indexOf(userId);
+        while (userIx !== -1) {
+          _entity._acl.r.splice(userIx, 1);
+          readers = _.get(_entity, ['_acl', 'r'], []);
+          userIx = readers.indexOf(userId);
+        }
+        return aclFunctions;
+      },
+      removeWriter: (userId) => {
+        let writers = _.get(_entity, ['_acl', 'w'], []);
+        let userIx = writers.indexOf(userId);
+        while (userIx !== -1) {
+          _entity._acl.w.splice(userIx, 1);
+          writers = _.get(_entity, ['_acl', 'w'], []);
+          userIx = writers.indexOf(userId);
+        }
+        return aclFunctions;
+      },
+      removeReaderGroup: (groupId) => {
+        let readers = _.get(_entity, ['_acl', 'groups', 'r'], []);
+        let groupIx = readers.indexOf(groupId);
+        while (groupIx !== -1) {
+          _entity._acl.groups.r.splice(groupIx, 1);
+          readers = _.get(_entity, ['_acl', 'groups', 'r'], []);
+          groupIx = readers.indexOf(groupId);
+        }
+        return aclFunctions;
+      },
+      removeWriterGroup: (groupId) => {
+        let writers = _.get(_entity, ['_acl', 'groups', 'w'], []);
+        let groupIx = writers.indexOf(groupId);
+        while (groupIx !== -1) {
+          _entity._acl.groups.w.splice(groupIx, 1);
+          writers = _.get(_entity, ['_acl', 'groups', 'w'], []);
+          groupIx = writers.indexOf(groupId);
+        }
+        return aclFunctions;
+      },
+      getGloballyReadable: () => {
+        if (typeof _entity._acl.gr === 'undefined') _entity._acl.gr = null;
```

Comment:
For all of these functions, the code previously made sure that `_entity._acl` was not null (by doing `_entity._acl?.gr`). Is that not actually needed (i.e. are we sure that `_entity._acl` always exists)?